client/context.go

```diff
@@ -0,0 +1,23 @@
+package client
+
+import (
+	"context"
+	"fmt"
+)
+
+type contextKey string
+
+func newContextKey(chainid string) contextKey {
+	return contextKey(fmt.Sprintf("chain-client-context/%s", chainid))
+}
+
```

Comment:
Comments on exported method

```suggestion
//SetChainClientOnContext sets ChainClient on Context
```